### PR TITLE
(PRE-94) Update README and make command line help less verbose

### PR DESCRIPTION
**Pull request by thallgren**
Updates the README file and shortens the command line help.

Details in the commit comments.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,13 +33,13 @@ The catalog_preview module is a Puppet Enterprise-only module that provides cata
...
 
-However, the `--migrate 3.8/4.0` option---which provides the specific migration checking that is the primary purpose of this module---can be used only when this module is used with a Puppet Enterprise <= 2015.2 version, **and** when the baseline environment uses current parser (Puppet 3) and the preview environment uses future parser (Puppet 4).
+However, the `--migrate 3.8/4.0` option---which provides the specific migration checking that is the primary purpose of this module---can be used only when this module is used with a Puppet Enterprise <= 2015.2 version, **and** when the baseline environment uses current parser (Puppet 3) and the preview environment uses future parser (Puppet 4). When no preview environment is given, the baseline environment will instead be compiled twice and the future parser will be enforced during the second compilation.
```
Comment (hlindberg) [inline README.md]: The reference to puppet version needs to include the FOSS version as well.

File: README.md [hunk 2]
```diff
@@ -77,20 +77,20 @@ Before you perform a migration preview, you should:
...
+When you run the preview compilation, you can turn on extra migration validation using `--migrate 3.8/4.0`. This turns on extra validations of future compatibility, flagging Puppet code that needs to be reviewed. This feature was introduced to help with the migration from the Puppet 3 parser to the Puppet 4 parser. The baseline environment must be configured to use the current (Puppet 3) parser and the `--preview-environment` must either reference an environment configured to use the future parser in its `environment.conf`, or it should not be given at all in which case the baseline environment will be compiled twice. Once with the Puppet 3 parser and once with the Puppet 4 parser.
 
 Note that the `--migrate 3.8/4.0` option is not available when using PE >= 2015.2.
```
Comment (hlindberg) [inline README.md]: This needs to also reference a FOSS version

Comment (thallgren) [inline README.md]: Which FOSS version should go in there?

File: README.md [hunk 3]
```diff
@@ -222,10 +228,10 @@ puppet preview --last --view overview --nodes diff_nodes
 ####Process output
 
 All output (except reports intended for human use) is written in JSON format to allow further processing with tools like 'jq' (JSON query). The output is written to a subdirectory named after the node of the directory appointed
-by the setting `preview_outputdir` (defaults to `$vardir/preview`):
+by the option `--preview-outputdir` (defaults to `$vardir/preview`):
```
Comment (hlindberg) [inline README.md]: IIRC, this is a setting - which is important to know since you can configure it instead of always giving it as an option.

Comment (thallgren) [inline README.md]: In that case, the documentation should say where this setting can be set. Is it in puppet.conf? If so, in what section?

File: lib/puppet_x/puppetlabs/preview/api/documentation/preview-help.md [hunk 4]
```diff
@@ -306,209 +189,3 @@ Note that all settings (such as 'log_level') affect both compilations.
...
-  bare words that are used in comparisons ==, >, <, >=, <=, matches =~ and !~, and when
-  used as case or selector options.
-
-  To fix these warnings, quote the upper case bare word if a string is intended, (or alter
-  the logic to use the type system in the unlikely event that the .3x. code did something in
-  relation to resource type name processing).
-
-COPYRIGHT
----------
-Copyright (c) 2015 Puppet Labs, LLC Licensed under Puppet Labs Enterprise.
```
Comment (hlindberg) [inline lib/puppet_x/puppetlabs/preview/api/documentation/preview-help.md]: Should this have a FOSS equivalent (c) ?

**Comment by hlindberg** [comment]
There is no easy way to get to the README.md. I think help needs to have a link to it when examples and details about errors are removed. An alternative is to let `--help readme` or something similar display the readme, or the URL to it (not sure if the readme.md works well if just displayed to the user).

File: README.md (hunk 3 above)
Comment (hlindberg) [inline README.md]: not sure which section - MAIN I guess, but preview may run in master mode, in which case it is MASTER. (I may work on both MAIN and MASTER). (The text is on line 457 (in PR)).

File: README.md (hunk 2 above)
Comment (hlindberg) [inline README.md]: 4.0.0, migrate only works on 3.x

**Comment by hlindberg** [comment]
Not sure how much effort to spend on that. It may be enough to say "For more detais and examples see the README.md" (but it would be nice if user then did not have to search for that on the file system).

File: lib/puppet_x/puppetlabs/preview/api/documentation/preview-help.md (hunk 4 above)
Comment (thallgren) [inline lib/puppet_x/puppetlabs/preview/api/documentation/preview-help.md]: Why should --help produce a copyright? It doesn't help much.

**Comment by thallgren** [comment]
My preference on what you're suggesting would be to use a separate `--help readme` but since that changes code, I think it deserves a separate issue (if we want to spend time on it, the readme is very visible at puppet forge and I don't think the user will have any problem finding it).

File: lib/puppet_x/puppetlabs/preview/api/documentation/preview-help.md (hunk 4 above)
Comment (hlindberg) [inline lib/puppet_x/puppetlabs/preview/api/documentation/preview-help.md]: (c) has meaning even in open source licenses, but nevermind...

**Pull request merged**